src/model.cpp

```diff
@@ -372,6 +373,32 @@ void Model::setParameterScale(std::vector<ParameterScaling> const& pscale) {
 std::vector<realtype> const& Model::getParameters() const {
     return originalParameters;
 }
+    
+realtype Model::getParameterById(std::string const& par_id) const {
+    if(!hasParameterIds())
+        throw AmiException("Could not access parameters by id as they are not set");
+    
+    auto par_ids = getParameterIds();
```

Comment:
Just to note: With the current implementation, the parameter vector is copied for each getParameterId call. This might be significant when iterating over thousands of parameters.

Effort could already be halved, by reusing par_ids instead of (the current implementation of) hasParameterIds.

Same for the other related functions.

Reply:
I don't like the idea of duplicating code for the sake of a meager 2 fold increase in efficency. I would guess that the compiler anyway inlines `!hasParameterIds()` detects that `getParameterIds()` is called twice.

Reply:
The duplication would be `par_ids.empty()`, I think that would be justifiable. ;) 
Not doing that won't kill us either.